integreat_cms/nominatim_api/nominatim_api_client.py

```diff
@@ -70,6 +71,9 @@ def search(
             raise RuntimeError(
                 "You can either specify query_str or pass additional keyword arguments, not both."
             )
+        if "street" in query_dict:
+            # If multiple street numbers are given, only take the first one
+            query_dict["street"] = re.sub(r"(\d+)-\d+", r"\1", query_dict["street"])
```

Comment:
I'm not completely sure whether I understand your concern correctly...
The purpose of this PR is not to validate whether an address format is correct, but to enable the auto completion for correct streets where a number range is given.
E.g. the street `Franziskanerstraße 6-8` is correct, but Nominatim did not return the correct result for it. This PR fixes this by converting it to `Franziskanerstraße 6` where Nominatim can find a correct result for.

Of course it's also possible to insert various symbols in the address which might or might not break the autocompletion feature, but this is completely unrelated to this PR in my opinion. But maybe I'm getting this wrong...

Reply:
> an address like `Franziskanerstr. 6,10-*!?234´<>` is valid

I don´t think it is - `\d+` only captures digits, after all:
![image](https://user-images.githubusercontent.com/47758554/197139928-8e4c611e-b889-4c93-af48-8d5e27e3e4a9.png)

However, the example `Franziskanerstr. 6,10-234` highlights two other issues:
![image](https://user-images.githubusercontent.com/47758554/197140414-acc04319-d773-4522-86c2-d2226fc78a7b.png)

Whitespace breaks the regex (unless it's stripped out somewhere beforehand?), and the while `6` is the actual first number, it's not being captured.

Maybe `(\d+)\s*[,-]\s*\d+` would do the trick?
![image](https://user-images.githubusercontent.com/47758554/197141435-1cf98619-2498-4d44-a9cf-0218422c05dc.png)

Reply:
@charludo ahh thanks, totally missed the possibility of other delimiters and spaces...
However, I think for e.g. `Franziskanerstraße 6, 10 - 234` we probably want `Franziskanerstraße 6` instead of `Franziskanerstraße 6 - 234`, right?
I tried to match a few more possible formats and hope I didn't miss some weird edge cases where this causes problems...

Reply:
> think for e.g. `Franziskanerstraße 6, 10 - 234` we probably want `Franziskanerstraße 6` instead of `Franziskanerstraße 6 - 234`, right?

Pretty sure in my examples above the capture group would only return the `6`, yes

Reply:
Hmm, not sure, do you mean like this?
```
In [2]: re.sub(r"(\d+)\s*[,-]\s*\d+", r"\1", "Franziskanerstraße 6, 10 - 234")
Out[2]: 'Franziskanerstraße 6 - 234'
```
I guess the problem is that the regex only matches two numbers separated by one delimiter and doesn't work with more than two numbers and multiple delimiters, right? I now tried to solve it like this:
https://github.com/digitalfabrik/integreat-cms/blob/5f5872d81c23476706a8899c5cc12f5e8dee722f/integreat_cms/nominatim_api/nominatim_api_client.py#L81

Reply:
@timoludwig oh sorry, I didn't look at it in context.

Reply:
@timoludwig oh sorry, I didn't look at it in context. So in this example,  `6, 10` is captured cy the entire regex, but `re.sub` only replaces it with the capture group (`6`). 

How about:
`re.sub(r"(\d+)(\s*[,-]\s*\d+)*", r"\1", "Franziskanerstraße 6, 10 - 234")`

Reply:
`re.compile(r"(\d+)").split("Franziskanerstraße 6, 10 - 234")`

might be a little more foolproof - however I believe there's countries where the street number goes in front of the street name? This solution would fail there, the previous one succeeds.

Reply:
> How about: `re.sub(r"(\d+)(\s*[,-]\s*\d+)*", r"\1", "Franziskanerstraße 6, 10 - 234")`

Ah, yeah looks good! That's very similar to the solution I now implemented, right? (I just added support for trailing letters and more delimiters)

```
# This expression matches a number optionally followed by a whitespace and one character
street_number = r"\d+( ?[a-zA-Z])?"
# This expression matches possible delimiters between multiple street numbers
delimiter = r" ?[/,\-–] ?"
# If multiple street numbers are given, only take the first one
query_dict["street"] = re.sub(
    rf"({street_number})({delimiter}{street_number})+",
    r"\1",
    query_dict["street"],
)
```

> `re.compile(r"(\d+)").split("Franziskanerstraße 6, 10 - 234")`
> 
> might be a little less errorprone - however I believe there's countries where the street number goes in front of the street name? This solution would fail there, the previous one succeeds.

I'm not sure, this even matches normal digits in the street name, e.g. "Straße des 17. Juni 123" or similar. So I guess we need the delimiter in the regex...

Reply:
Yeah, your solution looks very good to me. And a lot more readable than my oneliner 😄